src/inmanta/ast/type.py

```diff
@@ -590,7 +632,7 @@ class Literal(Union):
     """
 
     def __init__(self) -> None:
-        Union.__init__(self, [NullableType(Number()), Bool(), String(), TypedList(self), TypedDict(self)])
+        Union.__init__(self, [NullableType(Float()), Number(), Bool(), String(), TypedList(self), TypedDict(self)])
```

Comment:
This is the Union of all the types, By making one of them Nullable, we basically do "Float or None Or Bool or String..."
if we made them all Nullable it would be "Float or None Or Bool or None or String or None ..." So it comes down to the same. I moved it to Float as number will be gone one day.